src/main/java/org/apache/solr/benchmarks/indexing/UploadDocs.java

```diff
@@ -99,9 +102,12 @@ public void writeTo(OutputStream outstream) throws IOException {
 
         HttpResponse rsp = client.execute(httpPost);
         int statusCode = rsp.getStatusLine().getStatusCode();
-        if (statusCode != 200) {
+        if (!HttpStatus.isSuccess(statusCode)) {
             log.error("Failed a request: " +
                     rsp.getStatusLine() + " " + EntityUtils.toString(rsp.getEntity(), StandardCharsets.UTF_8));
+						if (interruptOnFailure) {
+							throw new RuntimeException("Failed to execute index call, status code: " + statusCode + " reason: " + rsp.getStatusLine().getReasonPhrase());
```

Comment:
A few comments on this area:
1. Can we get the hostname in the exception? I believe that would be `httpPost.getURI()`
2. Should this be an `IOException` instead of a `RuntimeException`?
3. There's some weird indenting here 😅

Reply:
1. Added the `updateEndpoint` as a part of the message
2. Good idea! Changed that to `IOException`
3. Yes, indeed weird indentation here 😁 , I took the liberty of reformatting the file to have 2 space as indentation adhering to fs-solr/solr standard